### PR TITLE
fix(ObjectArrayControl): fix double-default-item bug that happens in …

### PR DESCRIPTION
**Pull request by DrewHoo**
… strict mode

### CODE REVIEW

File: src/common/test-render.tsx [hunk 1]
```diff
@@ -1,8 +1,17 @@
...
 import { FormStateWrapper, RenderProps } from "./FormStateWrapper"
+import { StrictMode } from "react"
 
 export function render<T extends Record<string, unknown>, S>(
   props: RenderProps<T, S>,
 ) {
   return RTLrender(<FormStateWrapper {...props} />)
 }
+
+export function strictRender<T extends Record<string, unknown>, S>(
```
Comment (tyler-hoffman) [inline src/common/test-render.tsx:11]: The use case here is subtle. I would add a comment.

File: src/controls/ObjectArrayControl.tsx [hunk 2]
```diff
@@ -52,9 +51,12 @@ export function ObjectArrayControl({
...
   const prevDataValue = usePreviousValue(data)
-  if (data === 0 && prevDataValue === null) {
-    addItemToList()
-  }
+
+  useEffect(() => {
+    if (data === 0 && prevDataValue === null) {
+      addItemToList()
+    }
+  })
```
Comment (NathanFarmer) [inline src/controls/ObjectArrayControl.tsx:59]: I would love to figure out why we have inconsistent state issues between `ObjectArrayControl` and `PrimitiveArrayControl`. Primitive doesn't need this `useEffect`.

Comment (tyler-hoffman) [inline src/controls/ObjectArrayControl.tsx:59]: why no deps?

**Review by tyler-hoffman — APPROVED** [review]
The bug and why this fixes it is hard to get my head around. I think we need some kind of write-up or comment as to the "why" on how this works. My biggest question is around the lack of deps array with that useEffect.

**Review by NathanFarmer — APPROVED** [review]

**Pull request merged**